healthz/healthz.proto

```diff
@@ -50,24 +65,134 @@ message GetRequest {
+    google.protobuf.Any custom = 103;
+  }
+}
+
 message ComponentStatus {
   types.Path path = 1; // path of subcomponent.
+
   // Subcomponents that are aggregated by this status.
   repeated ComponentStatus subcomponents = 2;
+
   // Status of this component.
   Status status = 3;
+
   // Opaque data for how the healthcheck is implemented.  This can be any proto
   // defined by the vendor.  This could be the equivalent to outputs like
   // "show tech" or core files or any other diagnostic data.
-  google.protobuf.Any healthz = 4;
+  google.protobuf.Any healthz = 4 [deprecated=true];
+
+  // Artifacts provides links to all artifacts contained in this event.
+  // The individual artifacts can be retrieved via the Artifact() RPC.
+  repeated ArtifactHeader artifacts = 5;
+
+  // ID is the unique key for this event in the system.
+  string id = 6; 
+
+  // Acknowledged is set when a caller has processed the event.
+  bool acknowledged = 7;
+
+  // Created is the timestamp when this event was created.
+  google.protobuf.Timestamp created = 8;
+  
+  // Expires is the timestamp when the system will clean up the
+  // artifact. If unset, the artifact is not scheduled for GC.
+  google.protobuf.Timestamp expires = 9;
 }
 
-message GetResponse {
-  ComponentStatus component = 1;
+message ListRequest {
+  types.Path path = 1;
+  // By default only the unacknowledged events for the component will be
+  // returned.
+  bool include_acknowledged = 2;
+}
+
+message ListResponse {
+  repeated ComponentStatus status = 1;
+}
+
+message AcknowlegeRequest {
+  types.Path path = 1;
+  // Healthz event id.
+  string id = 2;
+}
+
+message AcknowledgeResponse {
+  ComponentStatus status = 1;
+}
+
+message ArtifactRequest {
+  // Artifact ID to be streamed.
+  string id = 1;
+}
+
+message ArtifactResponse {
+  oneof contents {
+    // Header is the first message in the stream. It contains
+    // the id of the artifact and metadata for the artifact
+    // based on the type of the artifact.
+    // OC defines FileArtifactType and ProtoArtifactType.
+    ArtifactHeader header = 1;
+    // Trailer contains the checksum for the streamed artifact.
+    Trailer trailer = 2;
+    bytes bytes = 3;    
+    google.protobuf.Any proto = 4;
+  }
+}
+
+message FileArtifactType {
+  // Local file name of the artifact.
+  string name = 1;
+  // Path to file on the local file system. (optional)
+  string path = 2;
+  // Mimetype of the file.
+  string mimetype = 3;
+  // Size of the file.
+  int64 size = 4;
+  // Hash of the file.
+  types.HashType hash = 5;
+}
+
+// Generic proto message artifact stream.
+// This proto tells the caller that the artifact stream 
+// will be a stream of proto encoded messages that make up
+// the artifact. Each message must be deserialized by the caller
+// and there are no other assumptions about the number of
+// messages or length of the stream or how those messages are to
+// be reassembled.
+message ProtoArtifactType {
```

Comment:
What was the motivation to have this rather than just use the `proto.Any`? Since this doesn't carry any additional information, do we expect that this is distinguishable at all from just having `bytes`?

Reply:
this is the header definition so a protoartifacttype has a clear definition on what it means
custom is custom in that it can define it's own encoding scheme past the defined proto schema which says
you must provide only a stream of proto messages 

vs. custom you could define whatever y our want

Reply:
Thanks for explaining, LGTM.